Create Totality.py

cmandrell opened the pull request.

castorf commented on capture/SharpCap/samples/Totality.py:
```diff
@@ -0,0 +1,129 @@
+
+exposure = ( 0.4, 4.0, 40.0, 400.0, 4000.0 ) # exposure tuple (ms)
+
+Time = True # Allow capture control by time
```
Let's not use capitalized names for flags, as the convention is to use this for classes.

I'd suggest "use_duration"  and "use_maxcount" , and rename number to maxcount.

and suppress_settings_file for Suppress.

castorf reviewed (changes requested):
Hey, I have a few comments on this... Would you mind addressing them before we push it back?

cmandrell commented:
Thanks for the input!

I will get this done later today. I have been working on a final today.

Chris

Sent from my Verizon, Samsung Galaxy smartphone
Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________
From: castorf ***@***.***>
Sent: Wednesday, May 10, 2023 3:15:09 PM
To: Dynamic-Eclipse-Broadcast-Initiative/debi ***@***.***>
Cc: Mandrell, Christopher T ***@***.***>; Author ***@***.***>
Subject: Re: [Dynamic-Eclipse-Broadcast-Initiative/debi] Create Totality.py (PR #4)


[EXTERNAL EMAIL ALERT]: Verify sender before opening links or attachments.

@castorf requested changes on this pull request.

Hey, I have a few comments on this... Would you mind addressing them before we push it back?

________________________________

In capture/SharpCap/samples/Totality.py<https://github.com/Dynamic-Eclipse-Broadcast-Initiative/debi/pull/4#discussion_r1190348566>:

> +
+import time
+#from datetime import datetime
+
+#date_form = datetime.now().strftime("%d_%b_%Y")
+
+### List of variables to be changed by user (User Input) #########################################
+path = r'C:\DEB\Exmouth\NoFilter\Totality' # path for folder
+#run_name = "bob"
+
+between_exp = 0.0 # time between each exposure (seconds)
+between_set = 0.0 # time between sets of exposures in exposure List (seconds)
+
+exposure = ( 0.4, 4.0, 40.0, 400.0, 4000.0 ) # exposure tuple (ms)
+
+Time = True # Allow capture control by time


Let's not use capitalized names for flags, as the convention is to use this for classes.

I'd suggest "use_duration" and "use_maxcount" , and rename number to maxcount.

and suppress_settings_file for Suppress.

________________________________

In capture/SharpCap/samples/Totality.py<https://github.com/Dynamic-Eclipse-Broadcast-Initiative/debi/pull/4#discussion_r1190349811>:

> +
+SharpCap.SelectedCamera.Controls.Binning.Value = '1'
+SharpCap.SelectedCamera.Controls.OutputFormat.Automatic = False
+SharpCap.SelectedCamera.Controls.OutputFormat.Value = 'TIFF files (*.tif)'
+SharpCap.SelectedCamera.Controls.ColourSpace.Value = 'MONO16'
+
+##########Opting for selecting from max of available resolutions instead of hardcoding value that might not exist
+#SharpCap.SelectedCamera.Controls.Resolution.Value = '3096x2078'
+SharpCap.SelectedCamera.Controls.Resolution.Value = SharpCap.SelectedCamera.Controls.Resolution.AvailableValues[0]
+
+SharpCap.SelectedCamera.Controls.Gain.Value = '50'
+
+######################################################################################
+
+### Capture images ####################################################################
+if ( Time ):


Note that in python unlike C you don't need parentheses here.

________________________________

In capture/SharpCap/samples/Totality.py<https://github.com/Dynamic-Eclipse-Broadcast-Initiative/debi/pull/4#discussion_r1190351574>:

> +##########Opting for selecting from max of available resolutions instead of hardcoding value that might not exist
+#SharpCap.SelectedCamera.Controls.Resolution.Value = '3096x2078'
+SharpCap.SelectedCamera.Controls.Resolution.Value = SharpCap.SelectedCamera.Controls.Resolution.AvailableValues[0]
+
+SharpCap.SelectedCamera.Controls.Gain.Value = '50'
+
+######################################################################################
+
+### Capture images ####################################################################
+if ( Time ):
+
+    start = time.time()
+    while ( (time.time() - start) <= duration):
+        Image(exposure, between_exp)
+        if Suppress:
+            SharpCap.Settings.CreateCameraSettingsFile = False


Given that the same code gets repeated here three times, may be better to always do the timekeeping and counting, and have a function which decides if we're still going.

—
Reply to this email directly, view it on GitHub<https://github.com/Dynamic-Eclipse-Broadcast-Initiative/debi/pull/4#pullrequestreview-1421309171>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AJURXQK2H2NSGOBVTA2IWH3XFPZM3ANCNFSM6AAAAAAX47AKXA>.
You are receiving this because you authored the thread.Message ID: ***@***.***>

cmandrell commented:
I have been thinking a bit more about this.

Since it seems like you wany a more polished product, I am going to add the naming convention for the file structure. I have the code, but I haven't tested it yet with a camera.

I will test tomorrow, and then I will get you the more finalized product.

Thanks for the input, again; I've never had anyone that actually does this for a living provide feedback for my code, so that is very good for me!

Chris

Sent from my Verizon, Samsung Galaxy smartphone
Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________
From: Mandrell, Christopher T ***@***.***>
Sent: Thursday, May 11, 2023 12:37:19 PM
To: Dynamic-Eclipse-Broadcast-Initiative/debi ***@***.***>; Dynamic-Eclipse-Broadcast-Initiative/debi ***@***.***>
Cc: Author ***@***.***>
Subject: Re: [Dynamic-Eclipse-Broadcast-Initiative/debi] Create Totality.py (PR #4)

Thanks for the input!

I will get this done later today. I have been working on a final today.

Chris

Sent from my Verizon, Samsung Galaxy smartphone
Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________
From: castorf ***@***.***>
Sent: Wednesday, May 10, 2023 3:15:09 PM
To: Dynamic-Eclipse-Broadcast-Initiative/debi ***@***.***>
Cc: Mandrell, Christopher T ***@***.***>; Author ***@***.***>
Subject: Re: [Dynamic-Eclipse-Broadcast-Initiative/debi] Create Totality.py (PR #4)


[EXTERNAL EMAIL ALERT]: Verify sender before opening links or attachments.

@castorf requested changes on this pull request.

Hey, I have a few comments on this... Would you mind addressing them before we push it back?

________________________________

In capture/SharpCap/samples/Totality.py<https://github.com/Dynamic-Eclipse-Broadcast-Initiative/debi/pull/4#discussion_r1190348566>:

> +
+import time
+#from datetime import datetime
+
+#date_form = datetime.now().strftime("%d_%b_%Y")
+
+### List of variables to be changed by user (User Input) #########################################
+path = r'C:\DEB\Exmouth\NoFilter\Totality' # path for folder
+#run_name = "bob"
+
+between_exp = 0.0 # time between each exposure (seconds)
+between_set = 0.0 # time between sets of exposures in exposure List (seconds)
+
+exposure = ( 0.4, 4.0, 40.0, 400.0, 4000.0 ) # exposure tuple (ms)
+
+Time = True # Allow capture control by time


Let's not use capitalized names for flags, as the convention is to use this for classes.

I'd suggest "use_duration" and "use_maxcount" , and rename number to maxcount.

and suppress_settings_file for Suppress.

________________________________

In capture/SharpCap/samples/Totality.py<https://github.com/Dynamic-Eclipse-Broadcast-Initiative/debi/pull/4#discussion_r1190349811>:

> +
+SharpCap.SelectedCamera.Controls.Binning.Value = '1'
+SharpCap.SelectedCamera.Controls.OutputFormat.Automatic = False
+SharpCap.SelectedCamera.Controls.OutputFormat.Value = 'TIFF files (*.tif)'
+SharpCap.SelectedCamera.Controls.ColourSpace.Value = 'MONO16'
+
+##########Opting for selecting from max of available resolutions instead of hardcoding value that might not exist
+#SharpCap.SelectedCamera.Controls.Resolution.Value = '3096x2078'
+SharpCap.SelectedCamera.Controls.Resolution.Value = SharpCap.SelectedCamera.Controls.Resolution.AvailableValues[0]
+
+SharpCap.SelectedCamera.Controls.Gain.Value = '50'
+
+######################################################################################
+
+### Capture images ####################################################################
+if ( Time ):


Note that in python unlike C you don't need parentheses here.

________________________________

In capture/SharpCap/samples/Totality.py<https://github.com/Dynamic-Eclipse-Broadcast-Initiative/debi/pull/4#discussion_r1190351574>:

> +##########Opting for selecting from max of available resolutions instead of hardcoding value that might not exist
+#SharpCap.SelectedCamera.Controls.Resolution.Value = '3096x2078'
+SharpCap.SelectedCamera.Controls.Resolution.Value = SharpCap.SelectedCamera.Controls.Resolution.AvailableValues[0]
+
+SharpCap.SelectedCamera.Controls.Gain.Value = '50'
+
+######################################################################################
+
+### Capture images ####################################################################
+if ( Time ):
+
+    start = time.time()
+    while ( (time.time() - start) <= duration):
+        Image(exposure, between_exp)
+        if Suppress:
+            SharpCap.Settings.CreateCameraSettingsFile = False


Given that the same code gets repeated here three times, may be better to always do the timekeeping and counting, and have a function which decides if we're still going.

—
Reply to this email directly, view it on GitHub<https://github.com/Dynamic-Eclipse-Broadcast-Initiative/debi/pull/4#pullrequestreview-1421309171>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AJURXQK2H2NSGOBVTA2IWH3XFPZM3ANCNFSM6AAAAAAX47AKXA>.
You are receiving this because you authored the thread.Message ID: ***@***.***>

JonathanMangin commented:
Awesome Christopher!I’m happy to try it out as well tomorrow if I have something to look at. Sent from my iPhoneOn May 12, 2023, at 6:16 PM, cmandrell ***@***.***> wrote:﻿
I have been thinking a bit more about this.

Since it seems like you wany a more polished product, I am going to add the naming convention for the file structure. I have the code, but I haven't tested it yet with a camera.

I will test tomorrow, and then I will get you the more finalized product.

Thanks for the input, again; I've never had anyone that actually does this for a living provide feedback for my code, so that is very good for me!

Chris

Sent from my Verizon, Samsung Galaxy smartphone
Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________
From: Mandrell, Christopher T ***@***.***>
Sent: Thursday, May 11, 2023 12:37:19 PM
To: Dynamic-Eclipse-Broadcast-Initiative/debi ***@***.***>; Dynamic-Eclipse-Broadcast-Initiative/debi ***@***.***>
Cc: Author ***@***.***>
Subject: Re: [Dynamic-Eclipse-Broadcast-Initiative/debi] Create Totality.py (PR #4)

Thanks for the input!

I will get this done later today. I have been working on a final today.

Chris

Sent from my Verizon, Samsung Galaxy smartphone
Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________
From: castorf ***@***.***>
Sent: Wednesday, May 10, 2023 3:15:09 PM
To: Dynamic-Eclipse-Broadcast-Initiative/debi ***@***.***>
Cc: Mandrell, Christopher T ***@***.***>; Author ***@***.***>
Subject: Re: [Dynamic-Eclipse-Broadcast-Initiative/debi] Create Totality.py (PR #4)


[EXTERNAL EMAIL ALERT]: Verify sender before opening links or attachments.

@castorf requested changes on this pull request.

Hey, I have a few comments on this... Would you mind addressing them before we push it back?

________________________________

In capture/SharpCap/samples/Totality.py<https://github.com/Dynamic-Eclipse-Broadcast-Initiative/debi/pull/4#discussion_r1190348566>:

> +
+import time
+#from datetime import datetime
+
+#date_form = datetime.now().strftime("%d_%b_%Y")
+
+### List of variables to be changed by user (User Input) #########################################
+path = r'C:\DEB\Exmouth\NoFilter\Totality' # path for folder
+#run_name = "bob"
+
+between_exp = 0.0 # time between each exposure (seconds)
+between_set = 0.0 # time between sets of exposures in exposure List (seconds)
+
+exposure = ( 0.4, 4.0, 40.0, 400.0, 4000.0 ) # exposure tuple (ms)
+
+Time = True # Allow capture control by time


Let's not use capitalized names for flags, as the convention is to use this for classes.

I'd suggest "use_duration" and "use_maxcount" , and rename number to maxcount.

and suppress_settings_file for Suppress.

________________________________

In capture/SharpCap/samples/Totality.py<https://github.com/Dynamic-Eclipse-Broadcast-Initiative/debi/pull/4#discussion_r1190349811>:

> +
+SharpCap.SelectedCamera.Controls.Binning.Value = '1'
+SharpCap.SelectedCamera.Controls.OutputFormat.Automatic = False
+SharpCap.SelectedCamera.Controls.OutputFormat.Value = 'TIFF files (*.tif)'
+SharpCap.SelectedCamera.Controls.ColourSpace.Value = 'MONO16'
+
+##########Opting for selecting from max of available resolutions instead of hardcoding value that might not exist
+#SharpCap.SelectedCamera.Controls.Resolution.Value = '3096x2078'
+SharpCap.SelectedCamera.Controls.Resolution.Value = SharpCap.SelectedCamera.Controls.Resolution.AvailableValues[0]
+
+SharpCap.SelectedCamera.Controls.Gain.Value = '50'
+
+######################################################################################
+
+### Capture images ####################################################################
+if ( Time ):


Note that in python unlike C you don't need parentheses here.

________________________________

In capture/SharpCap/samples/Totality.py<https://github.com/Dynamic-Eclipse-Broadcast-Initiative/debi/pull/4#discussion_r1190351574>:

> +##########Opting for selecting from max of available resolutions instead of hardcoding value that might not exist
+#SharpCap.SelectedCamera.Controls.Resolution.Value = '3096x2078'
+SharpCap.SelectedCamera.Controls.Resolution.Value = SharpCap.SelectedCamera.Controls.Resolution.AvailableValues[0]
+
+SharpCap.SelectedCamera.Controls.Gain.Value = '50'
+
+######################################################################################
+
+### Capture images ####################################################################
+if ( Time ):
+
+    start = time.time()
+    while ( (time.time() - start) <= duration):
+        Image(exposure, between_exp)
+        if Suppress:
+            SharpCap.Settings.CreateCameraSettingsFile = False


Given that the same code gets repeated here three times, may be better to always do the timekeeping and counting, and have a function which decides if we're still going.

—
Reply to this email directly, view it on GitHub<https://github.com/Dynamic-Eclipse-Broadcast-Initiative/debi/pull/4#pullrequestreview-1421309171>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AJURXQK2H2NSGOBVTA2IWH3XFPZM3ANCNFSM6AAAAAAX47AKXA>.
You are receiving this because you authored the thread.Message ID: ***@***.***>


—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you are subscribed to this thread.Message ID: ***@***.***>

castorf commented on capture/SharpCap/samples/Totality.py:
```diff
@@ -6,107 +6,80 @@
+#           Castor Fu
 # Written: 3/20/2023
-# Last Edit: 4/4/2023 CM
+# Last Edit: 5/15/2023 CM
```
Sorry, I should have said this earlier.  Since the revision control system tracks this stuff, there's no need to clutter up the code with it.  Please delete.

castorf commented on capture/SharpCap/samples/Totality.py:
```diff
@@ -125,5 +98,10 @@ def Image(exposure, between_exp):
                     Moved run_name variable to User input section
                     
-                    
+                  
```
These are better in your commit message.  That lets someone look over the revision history in github rather rthan digging into the code. 

The convention is one line summary / blank line / comments.

castorf reviewed (commented):
Hi, a couple more changes....

cmandrell commented:
I've never worked on GitHub before, so I will do my best here; not sure what some of that meant ; )

I will try again tomorrow.

Chris

Sent from my Verizon, Samsung Galaxy smartphone
Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________
From: castorf ***@***.***>
Sent: Tuesday, May 16, 2023 7:27:38 PM
To: Dynamic-Eclipse-Broadcast-Initiative/debi ***@***.***>
Cc: Mandrell, Christopher T ***@***.***>; Author ***@***.***>
Subject: Re: [Dynamic-Eclipse-Broadcast-Initiative/debi] Create Totality.py (PR #4)


[EXTERNAL EMAIL ALERT]: Verify sender before opening links or attachments.

@castorf commented on this pull request.

Hi, a couple more changes....

________________________________

In capture/SharpCap/samples/Totality.py<https://github.com/Dynamic-Eclipse-Broadcast-Initiative/debi/pull/4#discussion_r1195804554>:

>  # Written: 3/20/2023
-# Last Edit: 4/4/2023 CM
+# Last Edit: 5/15/2023 CM


Sorry, I should have said this earlier. Since the revision control system tracks this stuff, there's no need to clutter up the code with it. Please delete.

________________________________

In capture/SharpCap/samples/Totality.py<https://github.com/Dynamic-Eclipse-Broadcast-Initiative/debi/pull/4#discussion_r1195805818>:

> @@ -125,5 +98,10 @@ def Image(exposure, between_exp):
                     Added reset values for Exposure/Resolution/Gain for returning to live mode
                     Moved run_name variable to User input section

-
+


These are better in your commit message. That lets someone look over the revision history in github rather rthan digging into the code.

The convention is one line summary / blank line / comments.

—
Reply to this email directly, view it on GitHub<https://github.com/Dynamic-Eclipse-Broadcast-Initiative/debi/pull/4#pullrequestreview-1429613021>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AJURXQJGDO2XHNFMARJBIPLXGQLPVANCNFSM6AAAAAAX47AKXA>.
You are receiving this because you authored the thread.Message ID: ***@***.***>
